test/FlexVotingClient.invariants.t.sol

```diff
@@ -0,0 +1,183 @@
+  //   - sum of ProposalVote attr == weight of users expressing votes
+  //   - voting (without borrows) w/ flex client should not decrease vote weight
+
+  function test_multipleProposals() public {
+    assertEq(handler.proposalLength(), 0);
+    handler.propose("capital idea 'ol chap", 42424242);
+    assertEq(handler.proposalLength(), 1);
+    handler.propose("we should do dis", 1702);
+    assertEq(handler.proposalLength(), 2);
+    handler.propose("yuge, beautiful proposal", 5927392);
+    assertEq(handler.proposalLength(), 3);
+    handler.propose("a modest proposal", 1111111);
+    assertEq(handler.proposalLength(), 4);
+
+    // After 4 proposals we stop adding new ones. The call doesn't revert.
+    handler.propose("yessiree bob", 7777777);
+    assertEq(handler.proposalLength(), 4);
+  }
+
+  function test_deposit() public {
+    uint128 _amount = 42424242;
+
+    // It passes through deposits into the client.
+    assertEq(flexClient.deposits(address(this)), 0);
+    handler.deposit(_amount);
+    assertEq(flexClient.deposits(address(this)), _amount);
+    assertEq(handler.ghost_depositSum(), _amount);
+
+    // It enforces deposit maximums.
+    vm.expectPartialRevert(IERC20Errors.ERC20InsufficientBalance.selector);
+    handler.deposit(type(uint128).max);
+    assertEq(handler.ghost_depositSum(), _amount);
+    handler.deposit(type(uint128).max - _amount);
+    assertEq(handler.ghost_depositSum(), type(uint128).max);
+  }
+
+  function test_withdraw() public {
+    uint256 _userSeed = 1; // There's only one actor, so it doesn't matter.
+    uint128 _amount = 42424242;
+    handler.deposit(_amount);
+
+    // Deposits can be withdrawn from the flexClient through the handler.
+    handler.withdraw(_userSeed, _amount / 2);
+    assertEq(handler.ghost_depositSum(), _amount);
+    assertEq(handler.ghost_withdrawSum(), _amount / 2);
+    assertEq(flexClient.deposits(address(this)), _amount / 2);
+    assertEq(token.balanceOf(address(this)), _amount / 2);
+
+    handler.withdraw(_userSeed, _amount / 2);
+    assertEq(handler.ghost_withdrawSum(), _amount);
+    assertEq(token.balanceOf(address(this)), _amount);
+    assertEq(flexClient.deposits(address(this)), 0);
+  }
+
+  function test_withdrawAmountIsBounded() public {
+    uint256 _userSeed = 1; // There's only one actor, so it doesn't matter.
+    uint128 _amount = 42424242;
+    handler.deposit(_amount);
+
+    // Try to withdraw a crazy amount, it won't revert.
+    handler.withdraw(_userSeed, uint208(type(uint128).max));
+    assert(token.balanceOf(address(this)) > 0);
+    assert(token.balanceOf(address(this)) < _amount);
+    assert(flexClient.deposits(address(this)) > 0);
+    assert(flexClient.deposits(address(this)) < _amount);
+  }
+
+  function testFuzz_expressVote(
+    uint256 _userSeed,
+    uint256 _proposalId,
+    uint128 _amount
+  ) public {
+    _amount = uint128(bound(_amount, 1, type(uint128).max));
```

Comment:
In invariant tests, I prefer to use `_bound` instead of `bound`. The former does *not* log the result of the bound to console, and_for some reason—forge seems to include the invariant test bound results in stdout for invariant tests even if they don't fail.

Reply:
Didn't know about that. Done! Thanks